localstack/services/apigateway/templates.py

```diff
@@ -214,6 +214,12 @@ def render(
 
         variables = self.build_variables_mapping(api_context)
         result = self.render_vtl(template.strip(), variables=variables)
+
+        # set the request overrides into context
```

Comment:
basic support to header overrides just to support the use case, there is an issue open that will build on top of this. 

Example:

```
#set($content.requestOverride.header.X-Amz-Target = "PutEvents")
```